test/e2e/cypress/e2e/settings.cy.js

```diff
@@ -677,4 +677,133 @@ context('Settings page', () => {
+      // Also, it checks the retention time is displayed correctly
+      section()
+        .get('[aria-label="retention-time"]')
+        .invoke('text')
+        .as('initialRetentionTime');
+    });
+
+    describe.only('Changing Settings', () => {
+      it('should change retention time', () => {
+        // the initial retention time is read before the settings are changed
+        cy.get('@initialRetentionTime').then((text) =>
+          section()
+            .get('[aria-label="retention-time"]')
+            .should('have.text', text)
+        );
+
+        // Open the modal to change the settings
+        section().within(() => {
+          cy.get('button').contains('Edit Settings').click();
+        });
+
+        // Write the new retention time in the number selector
+        modal().get('input[role="spinbutton"]').clear().type('5', { delay: 0 });
+
+        // Save the new settings and wait for the response
+        cy.intercept('PUT', '/api/v1/settings/activity_log').as(
+          'changeSettings'
+        );
+        modal().get('button').contains('Save Settings').click();
+        cy.wait('@changeSettings');
+
+        // Check the new retention time is displayed correctly
+        section()
+          .get('[aria-label="retention-time"]')
+          .should('have.text', '5 months');
+      });
+
+      it('should not change settings when cancelling operation', () => {
+        // the initial retention time is read before the settings are changed
+        cy.get('@initialRetentionTime').then((text) =>
+          section()
+            .get('[aria-label="retention-time"]')
+            .should('have.text', text)
+        );
+
+        // Open the modal to change the settings
+        section().within(() => {
+          cy.get('button').contains('Edit Settings').click();
+        });
+
+        // Write some random value
+        modal()
+          .get('input[role="spinbutton"]')
+          .clear()
+          .type('999', { delay: 0 });
+
+        // Abort the operation
+        cy.intercept(
+          'PUT',
+          '/api/v1/settings/activity_log',
+          cy.spy().as('changeSettings')
+        );
+        modal().get('button').contains('Cancel').click();
+
+        // Check the modal is closed
+        cy.get('h3').should('not.have.text', 'Enter Activity Logs Settings');
+
+        // Check the data is not changed
+        cy.get('@initialRetentionTime').then((text) =>
+          section()
+            .get('[aria-label="retention-time"]')
+            .should('have.text', text)
+        );
+        cy.get('@changeSettings').should('not.have.been.called');
+      });
+
+      it('should return validation error on invalid input', () => {
+        // the initial retention time is read before the settings are changed
+        cy.get('@initialRetentionTime').then((text) =>
+          section()
+            .get('[aria-label="retention-time"]')
+            .should('have.text', text)
```

Comment:
Same as previous comments.

However I see the need for what we do later in line [801](https://github.com/trento-project/web/pull/2753/files#diff-2daf4a927dc90ea32c59092e95cbd3f017189a193fe4fca0fb2f6595dcf0ebb2R801)

Reply:
Yes exactly. I think is more about having a complete story in the test case. Pragmatically, it could be not very helpful.